uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/shared/controllers/request.controller.js

```diff
@@ -607,28 +607,51 @@ define([
      *
      * @return {Promise}
      */
-    function initRoles () {
+    function initRole () {
       role = 'staff';
 
-      // If the user is creating or editing their own leave, they will be
-      // treated as a staff regardless of their actual role.
-      if (vm.isSelfRecord) {
-        return;
-      }
+      return (vm.isSelfRecord
+        ? initRoleBasedOnSelfLeaveApproverState()
+        : initRoleBasedOnPermissions())
+        .finally(function () {
+          vm.canManage = vm.isRole('manager') || vm.isRole('admin');
```

Comment:
can we remove this from here, move it to a `initCanManage` function, and call that function from the `init` sequence after `initRole`?

Reply:
Done. I also checked if we need `finally`, I don't think so, so I changed to `then`